### PR TITLE
Security Hot Fix

### PR DESCRIPTION
**Pull request by dragonfly91**

### CODE REVIEW

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Providers/IaasVmPsBackupProvider.cs [hunk 1]
```diff
@@ -99,6 +100,12 @@ public BaseRecoveryServicesJobResponse EnableProtection()
                 containerUri = HelperUtils.GetContainerUri(keyValueDict, protectableObjectResource.Id);
                 protectedItemUri = HelperUtils.GetProtectableItemUri(
                     keyValueDict, protectableObjectResource.Id);
+
+                AzureIaaSVMProtectableItem iaasVmProteableItem = (AzureIaaSVMProtectableItem)protectableObjectResource.Properties;
```
Comment (pragrawa) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Providers/IaasVmPsBackupProvider.cs]: iaasVmProtectableItem

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.ServiceClientAdapter/Commands.RecoveryServices.Backup.ServiceClientAdapter.csproj [hunk 2]
```diff
@@ -97,6 +97,9 @@
       <SubType>Designer</SubType>
     </None>
   </ItemGroup>
+  <ItemGroup>
+    <Content Include="Microsoft.Azure.Management.RecoveryServices.Backup.dll" />
```
Comment (pragrawa) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.ServiceClientAdapter/Commands.RecoveryServices.Backup.ServiceClientAdapter.csproj]: Check  this, and remember to undo this temporary change before checkin.

Comment (dragonfly91) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.ServiceClientAdapter/Commands.RecoveryServices.Backup.ServiceClientAdapter.csproj]: Looks like the DLL was introduced in a previous PR and so this is not reflected in this PR.

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Providers/IaasVmPsBackupProvider.cs [hunk 3]
```diff
@@ -99,6 +100,12 @@ public BaseRecoveryServicesJobResponse EnableProtection()
                 containerUri = HelperUtils.GetContainerUri(keyValueDict, protectableObjectResource.Id);
                 protectedItemUri = HelperUtils.GetProtectableItemUri(
                     keyValueDict, protectableObjectResource.Id);
+
+                AzureIaaSVMProtectableItem iaasVmProtectableItem = (AzureIaaSVMProtectableItem)protectableObjectResource.Properties;
+                if (iaasVmProtectableItem != null)
+                {
+                    sourceResourceId = iaasVmProtectableItem.VirtualMachineId;
+                }
```
Comment (mkherani) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Providers/IaasVmPsBackupProvider.cs]: Check if else part is needed for logging or error handling

**Pull request merged**